float64 speed1
bool    mode1
float64 speed2
bool    mode2
